app/models/storage.rb

```diff
@@ -72,7 +72,8 @@ def to_s
   end
 
   def ext_management_systems
-    @ext_management_systems ||= Host.includes(:storages).select { |h| h.storages.include?(self) }.collect(&:ext_management_system).compact.uniq
+    @ext_management_systems ||= ExtManagementSystem.joins(:hosts => :storages).where(
+      :host_storages => {:storage_id => id}).uniq.to_a
```

Comment:
@agrare Can you review?  You made changes with respect to the join table recently.